### ISSUE TITLE
group errors by file

### ISSUE THREAD

**Issue by charlierudolph** [SimenB/stylint#409]
Currently you get an output like:
```
path/to/file1.styl
29 sortOrder error prefer alphabetical when sorting properties

path/to/file1.styl
33 sortOrder error prefer alphabetical when sorting properties

path/to/file1.styl
39:13 colons error unnecessary colon found

path/to/file1.styl
43 sortOrder error prefer alphabetical when sorting properties

path/to/file1.styl
44 sortOrder error prefer alphabetical when sorting properties

path/to/file1.styl
48 sortOrder error prefer alphabetical when sorting properties
```

while I would prefer output that looks like
```
path/to/file1.styl
29 sortOrder error prefer alphabetical when sorting properties
33 sortOrder error prefer alphabetical when sorting properties
39:13 colons error unnecessary colon found
43 sortOrder error prefer alphabetical when sorting properties
44 sortOrder error prefer alphabetical when sorting properties
48 sortOrder error prefer alphabetical when sorting properties
```

**Comment by SimenB** [comment]
This is possible today. https://github.com/SimenB/stylint/blob/master/README.md#groupoutputbyfile--default-true-true--false-

**Comment by charlierudolph** [comment]
Oh thanks. Good to know. 

One unexpected part of this is the following: I have a `.stylintrc` file that does not have the key `groupOutputByFile`, yet it is not defaulted to true. In my experience with other tools, if I don't specify a key, then it is given its default value. That would be preferable in the case.

**Comment by SimenB** [comment]
I agree

**Comment by PalleZingmark** [comment]
As far as I can see, this could be fixed by returning `true` on [Line 27 in `src/core/done.js`](https://github.com/SimenB/stylint/blob/master/src/core/done.js#L27) if `this.config.groupOutputByFile` doesn't exist. 
```javascript 
var group = this.config.groupOutputByFile || true 
``` 

I have a PR ready for this if agreed.

**Comment by charlierudolph** [comment]
@PalleZingmark I don't think that would work. If the person set it to false, that would override it to make it true. We only want to default it to true if the `groupOutputByFile` key is not present in the supplied config.

**Comment by PalleZingmark** [comment]
@charlierudolph Lol, of course, those damn _false javascript false being false_. ;) 

So I guess checking that the key actually exists would be a better way, e.g. 
```javascript
var group = ('groupOutputByFile' in this.config) ? this.config.groupOutputByFile : true 
```

**Comment by wojciechczerniak** [comment]
There is a default config already: https://github.com/SimenB/stylint/blob/master/src/core/config.js

Don't create another defaults in more places. This will be unmaintainable in the future.

The correct flow is to merge config files from various sources and use final value everywhere. Lets find why this fail first.

**Comment by PalleZingmark** [comment]
@wojciechczerniak True, good point. 

As far as I can see the config used is set in `src/utils/setConfig.js` based on X criterias, but as the description says: 
> @description overrides default config with a new config object

It only overrides the default config object with a new config object, hence non-existent props in config will return `false` and this problem will occur.

**Comment by PalleZingmark** [comment]
Changed my PR to use this approach instead of doing a check to whether the value exists or not. 
`setConfig` will now return a complete config based on [src/core/config.js](https://github.com/SimenB/stylint/blob/master/src/core/config.js), but with updated properties if provided. 

Need to update tests.
